src/Featurizers/AnalyticalRollingWindowFeaturizer.h

```diff
@@ -0,0 +1,305 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "../Archive.h"
+#include "Calculators/MeanCalculator.h"
+#include "Components/InferenceOnlyFeaturizerImpl.h"
+#include "Components/WindowFeaturizerBase.h"
+#include "Components/GrainFeaturizerImpl.h"
+#include "../Featurizer.h"
+#include "../Traits.h"
+
+#include <vector>
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+static constexpr char const * const         AnalyticalRollingWindowEstimatorName("AnalyticalRollingWindowEstimator");
+
+/////////////////////////////////////////////////////////////////////////
+///  \fn            AnalyticalRollingWindowCalculation
+///  \brief         Which type of calculation will be performed on the rolling window.
+///                 Currently only one of calculation is supported. This will be expanded upon in the future,
+///                 so we are peparing by making this an enum from the get go.
+///
+enum class AnalyticalRollingWindowCalculation : std::uint8_t {
+    Mean = 1
+};
+
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         AnalyticalRollingWindowTransformer
+///  \brief         RollingWindow class that is used for all analytical window computations.
+///
+template <
+    typename InputT,
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class AnalyticalRollingWindowTransformer:
+    public Components::InferenceOnlyTransformerImpl<InputT, std::vector<double>> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType = Components::InferenceOnlyTransformerImpl<InputT, std::vector<double>>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    AnalyticalRollingWindowTransformer(std::uint32_t windowSize, AnalyticalRollingWindowCalculation windowCalculation, std::uint32_t horizon, std::uint32_t minWindowCount = 1) ;

```

Comment:
>minWindowCount [](start = 156, length = 14)

minWindowSize

Reply:
Fixed. It was reverted by a git rebase, but its fixed again.